### PR TITLE
dhcp: Use upsteam .exmaple conf files and the once provided in the re…

### PR DESCRIPTION
**Pull request by krevsbech**
…cipe.

dhcp: Added conf packages for dhcp-server and dhcp-client

### CODE REVIEW

**Comment by kimrhh** [comment]
first line too long in commit ad36ff5
seems like the two next commits should be rebased out instead?

**Comment by kimrhh** [comment]
the whole thing needs to be rebased, to not first add stuff then remove it afterwards e.g. bind flags

**Comment by krevsbech** [comment]
Rebased as sugested

**Comment by KimBoendergaard** [comment]
Why put example files in 'config' packages? 
Keeping them in the 'main' package won't conflict with custom config packages like seen in other recipes (where two packages would otherwise provide the same files). But of course they'd take up space at target

**Comment by kimrhh** [comment]
since they are an example of how to generate a proper configuration file, i think it makes a lot of sense. You don't need the example on target, so when you override the config package you only have the working correct config.

**Comment by kimrhh** [comment]
also, we do not want weird empty packages, and we don't want to provide oe-lite.org configuration files, since there is no way of actually providing something that can be used downstream anyway. So when upstream have an example file (and no default conf file) we can provide that instead, and still allow a nice configuration override.

File: recipes/dhcp/dhcp4.inc [hunk 1]
```diff
@@ -53,41 +53,49 @@ do_install_extra_files () {
 		${D}${sysconfdir}/default/dhcp-relay
 	install -m 0755 ${SRCDIR}/init-server \
 		${D}${sysconfdir}/init.d/dhcp-server
-	install -m 0644 ${SRCDIR}/default-server \
+        install -m 0644 ${SRCDIR}/default-server \
```
Comment (kimrhh) [inline recipes/dhcp/dhcp4.inc]: please drop whitespace changes

File: recipes/dhcp/dhcp4.inc [hunk 2]
```diff
@@ -53,41 +53,49 @@ do_install_extra_files () {
...
 		${D}${sysconfdir}/default/dhcp-server
 
-	rm -f ${D}${sysconfdir}/dhclient.conf
-	rm -f ${D}${sysconfdir}/dhcpd.conf
-	install -m 0644 ${SRCDIR}/dhclient.conf \
-		${D}${sysconfdir}/dhcp/dhclient.conf
-	install -m 0644 ${SRCDIR}/dhcpd.conf \
-		${D}${sysconfdir}/dhcp/dhcpd.conf
+        mv ${D}${sysconfdir}/dhcpd.conf.example \
+    	   	${D}${sysconfdir}/dhcp/dhcpd.conf.example
```
Comment (kimrhh) [inline recipes/dhcp/dhcp4.inc]: seems like there are so tab/spaces issues here, make sure it is aligned with the current indention

File: recipes/dhcp/dhcp4.inc [hunk 3]
```diff
@@ -53,41 +53,49 @@ do_install_extra_files () {
...
+	mv ${D}${sysconfdir}/dhclient.conf.example \
+		${D}${sysconfdir}/dhcp/dhclient.conf.example
 
-	install -d ${D}${base_sbindir}/
+        install -d ${D}${base_sbindir}/
```
Comment (kimrhh) [inline recipes/dhcp/dhcp4.inc]: avoid whitespace changes

File: recipes/dhcp/dhcp4.inc [hunk 4]
```diff
@@ -56,38 +56,45 @@ do_install_extra_files () {
 	install -m 0644 ${SRCDIR}/default-server \
 		${D}${sysconfdir}/default/dhcp-server
 
-	rm -f ${D}${sysconfdir}/dhclient.conf
-	rm -f ${D}${sysconfdir}/dhcpd.conf
```
Comment (kimrhh) [inline recipes/dhcp/dhcp4.inc]: are these files no longer there?

File: recipes/dhcp/dhcp4.inc [hunk 5]
```diff
@@ -56,38 +56,45 @@ do_install_extra_files () {
...
-	install -m 0644 ${SRCDIR}/dhclient.conf \
-		${D}${sysconfdir}/dhcp/dhclient.conf
-	install -m 0644 ${SRCDIR}/dhcpd.conf \
-		${D}${sysconfdir}/dhcp/dhcpd.conf
-
+	mv ${D}${sysconfdir}/dhcpd.conf.example \
+	   	${D}${sysconfdir}/dhcp/dhcpd.conf.example
+	mv ${D}${sysconfdir}/dhclient.conf.example \
+	   	${D}${sysconfdir}/dhcp/dhclient.conf.example
 	install -d ${D}${base_sbindir}/
```
Comment (kimrhh) [inline recipes/dhcp/dhcp4.inc]: any reason to make these moves? can't we just package them from the original location?

**Comment by kimrhh** [comment]
looks good when we clear up the last small comments/questions

File: recipes/dhcp/dhcp4.inc (hunk 4 above)
Comment (krevsbech) [inline recipes/dhcp/dhcp4.inc]: No - They are not put there by upstream

File: recipes/dhcp/dhcp4.inc (hunk 5 above)
Comment (krevsbech) [inline recipes/dhcp/dhcp4.inc]: This is done to preserve how the package was placing the files before my changes.

**Comment by kimrhh** [comment]
merged, thanks

**Pull request merged**